authentication/pipeline/user_test.py

```diff
@@ -290,3 +288,36 @@ def test_require_micromasters_provider(
             user_actions.require_micromasters_provider(*args, **kwargs)
     else:
         assert user_actions.require_micromasters_provider(*args, **kwargs) == {}
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('backend_name,flow,social_auth,feature_flag_val,raises', [
+    ['email', SocialAuthState.FLOW_LOGIN, Mock(provider='saml'), True, True],
+    ['email', SocialAuthState.FLOW_LOGIN, Mock(provider='saml'), False, False],
+    ['email', SocialAuthState.FLOW_LOGIN, None, True, False],
+    ['email', SocialAuthState.FLOW_REGISTER, Mock(provider='saml'), True, False],
+    ['not_email', SocialAuthState.FLOW_LOGIN, Mock(provider='saml'), True, False],
+])
+def test_require_touchstone_login(
+        mocker, backend_name, flow, social_auth, feature_flag_val, raises
+):  # pylint: disable=too-many-arguments
+    """
+    Tests that verify test_require_touchstone_login raises a RequireProviderException
+    if (a) the user is logging in via email, (b) has a SAML-type authentication, and
+    (c) the SAML auth feature flag in on
+    """
+    settings.FEATURES[features.SAML_AUTH] = feature_flag_val
```

Comment:
Could you adjust this to use the `settings` fixture instead of the import so you get its rollback behavior and don't have this leak into other tests?